syncopy/io/load_ft.py

```diff
@@ -0,0 +1,419 @@
+    else:
+
+        if mem_use < 2000:
+            msg = "MAT-File version < 7.3 does not support lazy loading"
+            msg += f"\nReading {filename} might take up to 2GB of RAM, you requested only {mem_use / 1000}GB"
+            SPYWarning(msg)
+            
+        raw_dict = sio.loadmat(filename,
+                               mat_dtype=True,
+                               simplify_cells=True)
+
+        struct_keys = [skey for skey in raw_dict.keys() if '__' not in skey]
+
+        struct_container = raw_dict
+        struct_reader = lambda struct: _read_dict_structure(struct,
+                                                            add_fields=add_fields)
+
+    if len(struct_keys) == 0:
+        SPYValueError(legal="At least one structure",
+                      varname=filename,
+                      actual="No structure found"
+                      )
+
+    msg = f"Found {len(struct_keys)} structure(s): {struct_keys}"
+    SPYInfo(msg)
+
+    out_dict = {}
+
+    # load only a subset
+    if select_structures is not None:
+        keys = select_structures
+    # load all structures found
+    else:
+        keys = struct_keys
+
+    for skey in keys:
+        if skey not in struct_keys:
+            msg = f"Could not find structure `{skey}` in {filename}"
+            SPYWarning(msg)
+            continue
+
+        structure = struct_container[skey]
+        _check_req_fields(req_fields_raw, structure)
+        data = struct_reader(structure)
+        out_dict[skey] = data
+
+    return out_dict
+
+
+def _read_hdf_structure(h5Group,
+                        h5File,
+                        mem_use,
+                        add_fields=None):
+
+    '''
+    Each Matlab structure contained in
+    a hdf5 MAT-File is a h5py Group object.
+
+    Each key of this Group corresponds to
+    a field in the Matlab structure.
+
+    This is the translation from FT to Syncopy:
+
+    FT     Syncopy
+
+    label - channel
+    trial - trial
+    time  - time
+
+    optional:
+    fsample - samplerate
+
+    cfg - X
+
+    '''
+    # for user info
+    struct_name = h5Group.name[1:]
+
+    # this should be fixed upstream such that
+    # the `defaultDimord` is indeed the default :)
+    AData = AnalogData(dimord=AnalogData._defaultDimord)
+
+    # probably better to define an abstract mapping
+    # if we want to support more FT formats in the future
+
+    # these are numpy arrays holding hdf5 object references
+    # e.i. one per trial, channel, time (per trial)
+    trl_refs = h5Group['trial'][:, 0]
+    time_refs = h5Group['time'][:, 0]
+    chan_refs = h5Group['label'][0, :]
+
+    if 'fsample' in h5Group:
+        AData.samplerate = h5Group['fsample'][0, 0]
+    else:
+        AData.samplerate = _infer_fsample(h5File[time_refs[0]])
+
+    # -- retrieve shape information --
+    nTrials = trl_refs.size
+
+    # peek in 1st trial to determine single trial shape
+    # we only support equal trial lengths at this stage
```

Comment:
Makes total sense, however, I think some check or a more "controlled" try/except writing block to `ADset` might be warranted in case the trials do not have equal length. In its current form the code will crash with some cryptic HDF5 index error exception if any trial is longer/shorter than the first one